### ISSUE TITLE
Replace manual spec tables with specifications macro

### ISSUE THREAD

**Issue by queengooborg** [mdn/content#13126]
Now that https://github.com/mdn/yari/pull/5284 has landed, we can take advantage of the `{{Specifications}}` macro for entries that are not present in BCD.  We should convert all of our spec tables to the macro, either by linking to BCD where able or adding the `spec-urls` key to the page metadata.

<details>
<summary>
Breakdown of all files seemingly containing a manual spec table (found with <code>rg -i '\| Specification|<th scope="col">Specification</th>'</code>, last updated May 14, 2022):
</summary>

- /files/en-us/glossary/alpn/index.md
- /files/en-us/mdn/kitchensink/index.md
- /files/en-us/mdn/structures/page_types/aria_page_template/index.md
- /files/en-us/web/media/formats/containers/index.md
- /files/en-us/web/api/performance_api/index.md
- /files/en-us/web/api/css_object_model/index.md
- /files/en-us/web/api/web_storage_api/index.md
- /files/en-us/web/api/server-sent_events/index.md
- /files/en-us/web/api/page_visibility_api/index.md
- /files/en-us/web/api/svg_api/index.md
- /files/en-us/web/api/web_speech_api/index.md
- /files/en-us/web/api/decompressionstream/writable/index.md
- /files/en-us/web/api/sensor_apis/index.md
- /files/en-us/web/api/web_animations_api/index.md
- /files/en-us/web/api/pointer_lock_api/index.md
- /files/en-us/web/api/screen_capture_api/index.md
- /files/en-us/web/api/url_api/index.md
- /files/en-us/web/api/vibration_api/index.md
- /files/en-us/web/api/web_periodic_background_synchronization_api/index.md
- /files/en-us/web/api/notifications_api/index.md
- /files/en-us/web/api/storage_api/index.md
- /files/en-us/web/api/resource_timing_api/index.md
- /files/en-us/web/api/push_api/index.md
- /files/en-us/web/api/payment_request_api/index.md
- /files/en-us/web/api/web_midi_api/index.md
- /files/en-us/web/api/webrtc_statistics_api/index.md
- /files/en-us/web/api/websockets_api/index.md
- /files/en-us/web/api/permissions_api/index.md
- /files/en-us/web/api/subtlecrypto/digest/index.md
- /files/en-us/web/api/screen_wake_lock_api/index.md
- /files/en-us/web/api/presentation_api/index.md
- /files/en-us/web/api/picture-in-picture_api/index.md
- /files/en-us/web/api/webgl_api/index.md
- /files/en-us/web/api/streams_api/index.md
- /files/en-us/web/api/web_nfc_api/index.md
- /files/en-us/web/api/document_object_model/index.md
- /files/en-us/web/api/document_object_model/introduction/index.md
- /files/en-us/web/api/pointer_events/index.md
- /files/en-us/web/api/barcode_detection_api/index.md
- /files/en-us/web/api/indexeddb_api/index.md
- /files/en-us/web/api/webotp_api/index.md
- /files/en-us/web/api/reporting_api/index.md
- /files/en-us/web/api/web_workers_api/index.md
- /files/en-us/web/api/web_authentication_api/index.md
- /files/en-us/web/api/navigation_timing_api/index.md
- /files/en-us/web/api/performance_timeline/index.md
- /files/en-us/web/api/webrtc_api/index.md
- /files/en-us/web/api/web_crypto_api/index.md
- /files/en-us/web/api/webvtt_api/index.md
- /files/en-us/web/api/service_worker_api/index.md
- /files/en-us/web/api/background_fetch_api/index.md
- /files/en-us/web/http/network_error_logging/index.md
- /files/en-us/web/http/headers/accept-post/index.md
- /files/en-us/web/http/headers/sourcemap/index.md
- /files/en-us/web/http/headers/expect-ct/index.md
- /files/en-us/web/http/headers/proxy-authorization/index.md
- /files/en-us/web/http/resources_and_specifications/index.md
- /files/en-us/web/manifest/index.md
- /files/en-us/web/security/certificate_transparency/index.md
- /files/en-us/web/css/color_value/color_keywords/index.md
- /files/en-us/web/css/custom-ident/index.md
- /files/en-us/web/css/ident/index.md

**Comment by hamishwillee** [comment]
I'll help with this, but not until https://github.com/mdn/yari/pull/5284#issuecomment-1046350512 is answered.

**Comment by sideshowbarker** [comment]
This weekend I played around with automating this, and I have a reproducible working solution — based on a hack to the https://github.com/mdn/yari/blob/main/markdown/cli.ts source.

So I’ve gone ahead and assigned this to myself as a heads-up to preempt duplication of effort.

One side effect of the automation is that it causes redundant spaces to get reduced to a single space. I think that’s an expected consequence of the fact that what I’m doing is, first running `yari md m2h` to generate temporary `.html` files from the `.md` sources, and then running `yari md h2m` — with my hack in place — to replace the `.md` source with the output containing the `spec-urls` frontmatter key, and the specifications table replaced with `{{Specifications}}`.

In other words, it’s `.md`→`.html`→`.md` round-tripping the sources — and so any redundant spaces kind of expectedly get eaten during the initial `.md`→`.html` conversion (because the extra spaces anyway don’t get rendered in HTML output).

I guess it’s not a bad thing to zap all those extra spaces — but the problem is, it makes it harder to wade through the diffs to review the actual `spec-urls`-frontmatter-key-plus-`{{Specifications}}` change we actually care about.

So I think, to create the PRs, I’ll do some PRs that first just make the whitespace changes — and then after those are merged, I can run the conversion that’ll give us just the `spec-urls`-frontmatter-key-plus-`{{Specifications}}` changes to review.

**Comment by sideshowbarker** [comment]
> So I think, to create the PRs, I’ll do some that first just make the whitespace changes — and then after those are merged, I can run the conversion that’ll give us just the `spec-urls`-frontmatter-key-plus-`{{Specifications}}` changes to review.

Nevermind – after spending some more time thinking about it, I realized I could do some git gymnastics: the first commit with the unrelated (whitespace, etc.) changes, and then a second commit with the actual spec-URL changes we care about — and then rebasing and deleting the first commit. https://github.com/mdn/content/pull/13170 is the result.

So at this point, I’ll stop and write up the steps I used for doing the conversion, and clean up my https://github.com/mdn/yari/blob/main/markdown/cli.ts hack and open a yari PR with the patch — not because we’ll necessarily want to ever merge that into yari, but instead so that anybody else interested could try it out locally themselves.

**Comment by sideshowbarker** [comment]
https://github.com/mdn/yari/pull/5309 has the modification I made to the `h2m` command-line tool.

**Comment by teoli2003** [comment]
Note that web/http has a lot of _Unknown specification_ due to RFCs not in browser-specs (and some are Webdav).

**Comment by sideshowbarker** [comment]
> Note that web/http has a lot of _Unknown specification_ due to RFCs not in browser-specs (and some are Webdav).

Yup — see my comment over at https://github.com/mdn/content/pull/13170#discussion_r812657490. I think this part may be pretty easy to solve. Anyway, I’ll soon have something we can try out for it, at least.

**Comment by sideshowbarker** [comment]
As far as I can tell, this is completed. But if I’ve missed anything, please let me know
